src/reanimated2/animation/spring.ts

```diff
@@ -38,44 +53,91 @@ export function withSpring(
 
   return defineAnimation<SpringAnimation>(toValue, () => {
     'worklet';
-
-    // TODO: figure out why we can't use spread or Object.assign here
-    // when user config is "frozen object" we can't enumerate it (perhaps
-    // something is wrong with the object prototype).
-    const config: Required<SpringConfig> = {
+    const defaultConfig: Record<keyof SpringConfig, any> = {
       damping: 10,
       mass: 1,
       stiffness: 100,
       overshootClamping: false,
       restDisplacementThreshold: 0.01,
       restSpeedThreshold: 2,
       velocity: 0,
-    };
-    if (userConfig) {
-      Object.keys(userConfig).forEach(
-        (key) =>
-          ((config as any)[key] = userConfig[key as keyof typeof userConfig])
-      );
+      duration: 2000,
+      dampingRatio: 0.5,
+    } as const;
+
+    const config = { ...defaultConfig, ...userConfig };
+
+    // TODO this could be done only once
+    function initialCalculations(newMass = 0): {
```

Comment:
If this only depends on the mass maybe we should run it in onStart and keep omegas + zeta parameters in the config instead of having mass there.